src/Workspaces/Core/Portable/Utilities/BKTree.cs

```diff
@@ -97,8 +97,28 @@ public IList<string> Find(string value, int? threshold = null)
             }
         }
 
-        private void Lookup(Node currentNode, char[] queryCharacters, int queryLength, int threshold, List<string> result)
+        private void Lookup(
+            Node currentNode,
+            char[] queryCharacters,
+            int queryLength,
+            int threshold,
+            List<string> result, 
+            int recursionCount)
         {
+            // Don't bother recursing too deeply in the case of pathological trees.
+            // This really only happens when the actual code is strange (like
+            // 10,000 symbols all a single letter long).  In htat case, searching
+            // down this path will be fairly fruitless anyways.
+            //
+            // Note: this won't affect good searches against good data even if this
+            // pathological chain exists.  That's because the good items will still
+            // cluster near the root node in the tree, and won't be off the end of
+            // this long chain.
+            if (recursionCount > 256)
```

Comment:
Constant for easier tuning.  Also, is there a more deterministic way to decide if we're about to SO?